src/app/api/auth/onboarding/route.ts

```diff
@@ -0,0 +1,28 @@
+import { NextRequest, NextResponse } from "next/server";
+
+import { auth0 } from "@/node-lib/auth/auth0";
+import { userManagementClient } from "@/node-lib/auth/userManagement";
+import OakError from "@/errors/OakError";
+
+export const POST = auth0.withApiAuthRequired(async (req: NextRequest) => {
+  const res = new NextResponse();
+  const session = await auth0.getSession(req, res);
+
+  if (!session) {
+    // This shouldn't be reachable unless we've misconfigured the route middleware/decorator
+    throw new OakError({ code: "auth/missing-session", meta: { req } });
+  }
+
+  const appMetadata = { "owa:onboarded": true };
```

Comment:
We can add extra data here from the onboarding form to mark the user as a teacher. For now it simply flips the user to the onboarded state.